### ISSUE TITLE
Audio for IOS is coming out of ear speaker and not main speakers. 

### ISSUE THREAD

**Issue by JimAdamsdev** [NateRickard/Plugin.AudioRecorder#52]
When I play audio the sound just comes out of the ear speaker for IOS.  Can I do something to change this to come out of all the speakers or switch to the main speaker?  Thanks in advance for your help.  

Jim

**Comment by fliot4u** [comment]
did you find a solution for this. I am having the same exact problem. It changes all the audio output of my app to the ear speaker. I wonder if you found a work around to this issue. Thanks.

**Comment by JimAdamsdev** [comment]
Have not found a work around.  I did notice that if you close the app and then reopen it then all speakers

work.  However, when you record it switches it back.  I got pulled in a different direction.  Going to get back into

this later this week.  Wondered if a way exists to reset speakers after recording finishes.  Let me know if you have

any luck.  Thanks.

**Comment by fliot4u** [comment]
I did found a solution or workaround. 
Create an interface on your share project like this:
namespace <yournamespace>.IOS
{
    public interface IAudioService
    {
    }
}

and create a class AudioService.cs that implements it just like this:
using System;
using AVFoundation;
using Foundation;

namespace <yournamespace>.iOS
{
    public class AudioService : IAudioService
    {
        public AudioService()
        {
        }

        public void PlaySoundThroughSpeaker()
        {
            var session = AVAudioSession.SharedInstance();
            session.SetCategory(AVAudioSessionCategory.Playback);
            session.SetActive(true);

        }
        public void PlaySoundThroughEarPiece()
        {
            var session = AVAudioSession.SharedInstance();
            session.SetCategory(AVAudioSessionCategory.PlayAndRecord);
            session.SetActive(true);
            

        }
    }
}

then in your event clicked do something like this:
 void RecordClicked(System.Object sender, System.EventArgs e)
        {
			var audioService = new AudioService();
			audioService.PlaySoundThroughEarPiece();
			RecordAudio();
	}
because before recording we have to allow the phone to use the microphone and ear speaker.
and then in wherever you handle the audio file created by the recorder add this code at the end:

private void Recorder_AudioInputReceived(object sender, string audioFile)
        {
			
			DisplayAlert("Notification", "Audio Message Sent Succesfully!", "Close");

			
			var audioService = new AudioService();
			audioService.PlaySoundThroughSpeaker();

		}

basically we are changing from the main speakers to the ear speakers. but as I notice if we are going to record we need to set it as ear speakers for that but we can easily change it back when we are done recording. Hope it helps! Good Luck!

**Comment by JimAdamsdev** [comment]
Thanks! 😊  My android phone has only one speaker.  Have you tested to see if its also occuring on Android?

**Comment by fliot4u** [comment]
At least for me the android side doesn't give any issues.
But in order to run that code in the shared  code you can either add a try{}catch{} or put the code inside a:
 #if __IOS__ 
var audioService = new AudioService();
audioService.PlaySoundThroughSpeaker();
#endif
because android will give you an exception when you try to initialize an audio service.

**Comment by grockland** [comment]
#if __IOS__ and implementing a AudioService in Xamarin iOS for a DependencyInjection with much the same code as above (fliot4u)  worked for me.  Android ok out of the box.
